### PR TITLE
Upgrade to lsstts/develop-env:c0018.000

### PR DESCRIPTION
**Pull request by sebastian-aranda**
A new development environment was released, so this PR updates the image and make some adjustments to comply with the new version. Also some code refactor and linter fixes were done.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ pipeline {
     dockerImageName = "lsstts/love-producer:"
     dockerImage = ""
     dockerLoveCSCImageName = "lsstts/love-csc:"
-    LSSTTS_DEV_VERSION = "c0016.001"
+    LSSTTS_DEV_VERSION = "c0018.000"
```
Comment (areyesd14) [inline Jenkinsfile:8]: In this line, It should be dev-cycle variable?

Comment (sebastian-aranda) [reply Jenkinsfile:8]: Thanks for the review @areyesd14 ! Well, this is part of the things that we haven't extended yet. The Jenkinsfile can't read the .env file with the respective environment variables, opposed to the docker-compose that implicit loads the environment variables of the .env file. So no, it has to be hardcoded until we find a method to extend this.

Comment (areyesd14) [reply Jenkinsfile:8]: Ok, that makes sense.

File: producer/csc/test_client.py [hunk 2]
```diff
@@ -19,7 +20,7 @@ async def test_valid_remote_not_in_config(self):
         async def arrange():
             from csc.client import CSCWSClient
 
-            salobj.set_random_lsst_dds_domain()
+            salobj.set_random_lsst_dds_partition_prefix()
```
Comment (areyesd14) [inline producer/csc/test_client.py:23]: What is the difference between this line and the line of the before version?

Comment (sebastian-aranda) [reply producer/csc/test_client.py:23]: The `set_random_lsst_dds_domain` was deprecated and replaced for `set_random_lsst_dds_partition_prefix`. Here is the documentation link: https://ts-salobj.lsst.io/py-api/lsst.ts.salobj.set_random_lsst_dds_domain.html

Comment (areyesd14) [reply producer/csc/test_client.py:23]: Great! ;)

**Review by areyesd14 — APPROVED** [review]
This was a great refactor!
Great job! Thanks a lot!

**Pull request merged**